python/dgl/distributed/dist_context.py

```diff
@@ -167,16 +172,20 @@ def submit_task_to_all_workers(self, func, args):
 
     def get_result(self, dataloader_name, timeout=1800):
         """Get result from result queue"""
-        result_dataloader_name, result = self.result_queue.get(timeout=timeout)
-        assert result_dataloader_name == dataloader_name
-        return result
+        if dataloader_name not in self.results:
+            raise DGLError(f"{dataloader_name} not found.")
+        while len(self.results[dataloader_name]) == 0:
+            dl_name, data = self.result_queue.get(timeout=timeout)
```

Comment:
What will happen if `self.result_queue.get` timed out? Are we able to catch that?

Reply:
raises the [queue.Empty](https://docs.python.org/3/library/queue.html#queue.Empty) exception if no item was available within that time.
As this should never happen in normal runnings, we don't catch such exception and just let it abort the app. I think this behavior is acceptable. If it happens, something went wrong in the sampler processes.